joomla/joomla-cms
# [4.0] Fix for AccessibleMediaField

Pull Request for Issue # https://github.com/joomla/joomla-cms/issues/29825.

### Summary of Changes
If the `accessiblemedia` field within in a `subfield`, `value` is an object `stdClass`. I extended the check for this. Also, I added the unit tests for the field and do not use `$field->apivalue`any more because the name is misleading and it is not possible in a `subfield` custom field.


### Testing Instructions

1. Please test like explained here: https://github.com/joomla/joomla-cms/pull/27712

2. Please also check whether the media custom field can be used within a subform custom field. Note: If you insert the media custom field in a subform custom field, you can only select (when editing the article) an image file after you have saved the article. This issue is not related to this PR. Please see https://github.com/joomla/joomla-cms/issues/26711.

3. Check also the rendering in the front end for the custom field of type `media` and `subfield incluing media field`.

### Documentation Changes Required
Yes
